### PR TITLE
added alerts for missing MC deployments

### PR DESCRIPTION
**Pull request by AverageMarcus**
This PR:

- adds `ManagementClusterDeploymentMissingFirecracker` and `ManagementClusterDeploymentMissing` alerts added to notify on missing critical MC deployments.

Related issue: https://github.com/giantswarm/giantswarm/issues/18665

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).
- [ ] Alerting rules must have a comment documenting why it needs to exist.

### CODE REVIEW

File: helm/prometheus-rules/templates/alerting-rules/management-cluster.rules.yml [hunk 1]
```diff
@@ -128,3 +128,17 @@ spec:
         severity: notify
         team: ludacris
         topic: kubernetes
+    - alert: ManagementClusterDeploymentMissing
+      annotations:
+        description: '{{`Deployment {{ $labels.workload_name }} is missing.`}}'
+        opsrecipe: management-cluster-deployment-is-missing/
+      expr: absent(kube_deployment_status_condition{namespace="giantswarm", condition="Available", workload_name="management-cluster-admission-controller-manager"})
```
Comment (anvddriesch) [inline helm/prometheus-rules/templates/alerting-rules/management-cluster.rules.yml:135]: Is the name of the deployment actually `management-cluster-admission-controller-manager`? It does not ring any bell for me right now.

Comment (AverageMarcus) [reply helm/prometheus-rules/templates/alerting-rules/management-cluster.rules.yml:135]: Unless I've got the wrong deployment. There definitely is a deployment called `management-cluster-admission-controller-manager` on gauss

Comment (anvddriesch) [reply helm/prometheus-rules/templates/alerting-rules/management-cluster.rules.yml:135]: Ah okay. :) I guess it is not owned by firecracker. Maybe for now we can try with just the `aws-admission-controller`.

Comment (AverageMarcus) [reply helm/prometheus-rules/templates/alerting-rules/management-cluster.rules.yml:135]: Isn't it going to become owned by Phoenix though going forward? (Might be wrong)

Either way, happy to reduce the scope of this for now to just focus on the aws-admission-controller and then adding more when needed.

**Review by anvddriesch — APPROVED** [review]
If this is tested, let's go with this for now :)

**Comment by AverageMarcus** [comment]
Not firing:

CleanShot 2021-10-05 at 1 52 31

Pending:

CleanShot 2021-10-05 at 2 00 55

Firing:

CleanShot 2021-10-05 at 2 09 03

**Comment by anvddriesch** [comment]
Exciting stuff

**Pull request merged**